drivers/tee/optee/rpc.c

```diff
@@ -375,16 +430,19 @@ static void handle_rpc_func_cmd(struct tee_context *ctx, struct optee *optee,
 		handle_rpc_func_cmd_wait(arg);
 		break;
 	case OPTEE_MSG_RPC_CMD_SHM_ALLOC:
-		handle_rpc_func_cmd_shm_alloc(ctx, arg);
+		free_page_array(call_ctx);
```

Comment:
Contrary to what I've said before, it should be enough to call this function here (and of course from `optee_rpc_finalize_call()`) as not much is gained by calling it for all the other RPCs.